### ISSUE TITLE
InMemory creates new row for modified entries that do not exist

### ISSUE THREAD

**Issue by rowanmiller** [aspnet/EntityFramework#3953]
The code below gives the following output, meaning that a new row gets created for the modified state entry even though no row with that key exists.

```
Modified
22 - blogs.msdn.com/dotnet
Press any key to continue . . .
```

```c#
using Microsoft.Data.Entity;
using System;
using System.Linq;

namespace ConsoleApplication1
{
    class Program
    {
        static void Main(string[] args)
        {
            using (var db = new BloggingContext())
            {
                db.Blogs.Update(new Blog { Id = 22, Url = "blogs.msdn.com/dotnet" });
                Console.WriteLine(db.ChangeTracker.Entries().Single().State);
                db.SaveChanges();
            }

            using (var db = new BloggingContext())
            {
                foreach (var item in db.Blogs)
                {
                    Console.WriteLine($"{item.Id} - {item.Url}");
                }
            }
        }
    }

    public class Blog
    {
        public int Id { get; set; }
        public string Url { get; set; }
    }

    public class BloggingContext : DbContext
    {
        public DbSet<Blog> Blogs { get; set; }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            optionsBuilder.UseInMemoryDatabase();
        }
    }
}
```

**Comment by rowanmiller** [comment]
Make sure we also test the delete case from https://github.com/aspnet/EntityFramework/issues/4515

**Comment by ajcvickers** [comment]
Already fixed as part of #4680
